cmd/immuadmin/immuadmin.go

```diff
@@ -0,0 +1,220 @@
+	cmd.PersistentFlags().StringVar(&o.CfgFn, "config", "", "config file (default path are configs or $HOME. Default filename is immuclient.ini)")
+	cmd.PersistentFlags().BoolP("mtls", "m", client.DefaultOptions().MTLs, "enable mutual tls")
+	cmd.PersistentFlags().String("servername", client.DefaultMTLsOptions().Servername, "used to verify the hostname on the returned certificates")
+	cmd.PersistentFlags().String("certificate", client.DefaultMTLsOptions().Certificate, "server certificate file path")
+	cmd.PersistentFlags().String("pkey", client.DefaultMTLsOptions().Pkey, "server private key path")
+	cmd.PersistentFlags().String("clientcas", client.DefaultMTLsOptions().ClientCAs, "clients certificates list. Aka certificate authority")
+	if err := viper.BindPFlag("default.port", cmd.PersistentFlags().Lookup("port")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.address", cmd.PersistentFlags().Lookup("address")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.mtls", cmd.PersistentFlags().Lookup("mtls")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.servername", cmd.PersistentFlags().Lookup("servername")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.certificate", cmd.PersistentFlags().Lookup("certificate")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.pkey", cmd.PersistentFlags().Lookup("pkey")); err != nil {
+		return err
+	}
+	if err := viper.BindPFlag("default.clientcas", cmd.PersistentFlags().Lookup("clientcas")); err != nil {
+		return err
+	}
+	viper.SetDefault("default.port", gw.DefaultOptions().ImmudbPort)
+	viper.SetDefault("default.address", gw.DefaultOptions().ImmudbAddress)
+	viper.SetDefault("default.mtls", client.DefaultOptions().MTLs)
+	viper.SetDefault("default.servername", client.DefaultMTLsOptions().Servername)
+	viper.SetDefault("default.certificate", client.DefaultMTLsOptions().Certificate)
+	viper.SetDefault("default.pkey", client.DefaultMTLsOptions().Pkey)
+	viper.SetDefault("default.clientcas", client.DefaultMTLsOptions().ClientCAs)
+
+	return nil
+}
+
+func options() *client.Options {
+	port := viper.GetInt("default.port")
+	address := viper.GetString("default.address")
+	mtls := viper.GetBool("default.mtls")
+	certificate := viper.GetString("default.certificate")
+	servername := viper.GetString("default.servername")
+	pkey := viper.GetString("default.pkey")
+	clientcas := viper.GetString("default.clientcas")
+	options := client.DefaultOptions().
+		WithPort(port).
+		WithAddress(address).
+		WithAuth(true).
+		WithMTLs(mtls)
+	if mtls {
+		// todo https://golang.org/src/crypto/x509/root_linux.go
+		options.MTLsOptions = client.DefaultMTLsOptions().
+			WithServername(servername).
+			WithCertificate(certificate).
+			WithPkey(pkey).
+			WithClientCAs(clientcas)
+	}
+	return options
+}
+
+func optionsWithAuth() *client.Options {
+	var token string
```

Comment:
I believe this can be avoid thanks to the new constructor available in client: client.NewImmuClient(options())